src/scripts/migrate.py

```diff
@@ -0,0 +1,93 @@
+"""Slurp migration pipeline
+
+TODO's:
+  - add CLI: look to makefile for what to include
+"""
+import os
+from argparse import ArgumentParser
+from typing import Dict, List
+
+import oaklib
+import pandas as pd
+
+
+# TODO: implement this func:
+# todo: IDs should be int or str? prolly str
+def determine_next_available_mondo_id(min_id: str, mondo_termlist_df: pd.DataFrame) -> str:
+    """Starting from `min_id`, count up and check until finding the next ID."""
+    next_id = str(0)
+    return next_id
+
+
+def run(ontology_path: str, sssom_map_path: str, min_id: str, mondo_terms_path: str, outpath: str) -> pd.DataFrame:
+    """Run slurp pipeline for given ontology"""
+    # TODO: read this with OAK
+    source_ontology = ontology_path
+    sssom_df = pd.read_csv(sssom_map_path, comment='#', sep='\t')
+    # TODO: Need to get the mondo terms, but ran out of memory on my alternate PC. get from other PC.
+    print(f'exists: {mondo_terms_path}: ', os.path.exists(os.path.join(os.getcwd(), mondo_terms_path)))
+    # mondo_termlist_df = pd.read_csv(mondo_terms_path, comment='#', sep='\t')
+    mondo_termlist_df = pd.DataFrame()
+
+    source_onto_terms: List[str] = []
+    sssom_object_ids = set(sssom_df['object_id'])
+
+    data = []
+    for t in source_onto_terms:
+        if t not in sssom_object_ids:
+            migrate = True
+            # todo: Find the correct way of doing this:
+            parents: List[str] = oaklib.get_direct_parents(t)
+            for parent in parents:
+                if parent not in sssom_object_ids:
+                    migrate = False
+                    break
+                else:
+                    obj_data = sssom_df[sssom_df['object_id'] == parent]
+                    pred = str(obj_data['predicate_id'])
+                    if pred in ['skos:exactMatch', 'skos:narrowMatch']:
+                        # In other words, if the parent is mapped, and the mapping is either exact or narrower
+                        parents.append(obj_data['subject_id'])
+                    else:
+                        pass  # Its fine, just continue looking for other parents in this case
+            if migrate and parents:
+                # TODO: implement this func:
+                next_mondo_id = determine_next_available_mondo_id(min_id, mondo_termlist_df)  # satrting from min_id, then counting up and checking if it does not already exist.
+                # todo: Find the correct way of doing this:
+                label = oaklib.get_label(t)
+                # todo: Find the correct way of doing this:
+                definition = oaklib.get_definition(t)
+                data.append({'mondo_id': next_mondo_id, 'xref': t, 'label': label, 'definition': definition})
+
+    result = pd.DataFrame(data)
+    result.to_csv(outpath, sep="\t")
+    return result
+
+
+def cli():
+    """Command line interface."""
+    package_description = \
+        'Slurp pipeline: Integrate new terms from other ontologies into Mondo.'
```

Comment:
@Nico: If any of these explanations feel inaccurate to you, let me know what you'd like the text to be.

Reply:
Looks great so far!